pkg/actors/placement/placement.go

```diff
@@ -551,10 +551,10 @@ func (p *actorPlacement) updatePlacements(in *v1pb.PlacementTables) {
 			// TODO: @elena in v1.15 remove the check for versions < 1.13
 			// only keep `hashing.NewFromExisting`
 
-			if p.apiLevel < raft.NoVirtualNodesInPlacementTablesAPILevel {
-				entries[k] = hashing.NewFromExistingWithVirtNodes(v.GetHosts(), v.GetSortedSet(), loadMap)
-			} else {
+			if in.GetReplicationFactor() > 0 && len(v.GetHosts()) == 0 {
```

Comment:
Couldn’t len hosts be 0 in correct operation? I think it would make sense for placement to return a header which daprd checks for here. `dapr-content-vnodes=false`.

Reply:
Yes, `v.GetHosts()` will always be empty in correct operation in 1.13+. 
For 1.12 and lower, it should never be empty, because the disseminated placement table is dynamically generated based on the host data saved in Raft, meaning - if there is an entry for an actor type (`v`) it will always have a `load_map` and consequentially - `hosts` and `sorted_set`.

Reply:
>  if there is an entry for an actor type (v) it will always have a load_map and consequentially - hosts and sorted_set.

Doesn't this mean it will be empty when there is no actor type?}

Reply:
The placement table pre 1.13 looks like this:
```
{
  "entries": {
    "myactortype-0": {
      "hosts": {
        "10056481384176189962": "192.168.1.89:62362",
        "10100244799470048543": "192.168.1.89:62362",
        "10136976372505843558": "192.168.1.89:62362",
        "10210620769057790472": "192.168.1.89:62362",
        "10278294263954349758": "192.168.1.89:62345",
		    ...
        "9959649394905760010": "192.168.1.89:62345",
        "999880486694283580": "192.168.1.89:62362"
      },
      "sorted_set": [
        3430792224860206,
        10660957680821358,
        69520260237277928,
        96174763468296109,
        110417171835179485,
        ...
        18409429381221104921,
        18426863874365556280
      ],
      "load_map": {
        "192.168.1.89:62345": {
          "name": "192.168.1.89:62345",
          "id": "08aa8f86-df9b-45bd-b407-f35c67535073"
        },
        "192.168.1.89:62362": {
          "name": "192.168.1.89:62362",
          "id": "140d66c3-66b4-4642-ade5-0d17b4cd0402"
        },
        "192.168.1.89:62367": {
          "name": "192.168.1.89:62367",
          "id": "6c1f823e-b3b0-43d6-b7ca-ce844ae1ae2c"
        }
      }
    },
    "myactortype-1": {
      "hosts": {
        "10056481384176189962": "192.168.1.89:62362",
        "10100244799470048543": "192.168.1.89:62362",
        "10136976372505843558": "192.168.1.89:62362",
        ...
        "9475149095182223555": "192.168.1.89:62362",
        "999880486694283580": "192.168.1.89:62362"
      },
      "sorted_set": [
        69520260237277928,
        110417171835179485,
       ...
        18297525576440347375,
        18375786333053603218
      ],
      "load_map": {
        "192.168.1.89:62362": {
          "name": "192.168.1.89:62362",
          "id": "140d66c3-66b4-4642-ade5-0d17b4cd0402"
        }
      }
    }
  },
  "version": "3",
  "api_level": 20
}
```

If an actor type is not present, it won't be a part of the placement table in the first place.

Reply:
OK. It seems a bit flaky to me to rely on the data structure contents to make this decision, rather than the same API surface which we use to signal whether to return the vnodes. If we are sure that this will never be problem, than fine.

Reply:
I agree with @elena-kolevska. I like that we use the content to decide here.